pkg/aiapiclient/aiapiclient.go

```diff
@@ -29,14 +36,85 @@ func NewAiAPIClient() AiAPIClient {
 	defaultAppInsightsAppID := os.Getenv("APP_INSIGHTS_APP_ID")
 	appInsightsKey := os.Getenv("APP_INSIGHTS_KEY")
 
+	// if no application insights key has been specified, then we will use AD authentication
 	return AiAPIClient{
-		appID:  defaultAppInsightsAppID,
-		appKey: appInsightsKey,
+		appID:           defaultAppInsightsAppID,
+		appKey:          appInsightsKey,
+		useADAuthorizer: appInsightsKey == "",
 	}
 }
 
 // GetMetric calls to API to retrieve a specific metric
-func (ai AiAPIClient) GetMetric(metricInfo MetricRequest) (*MetricsResponse, error) {
+func (ai AiAPIClient) GetMetric(metricInfo MetricRequest) (*insights.MetricsResult, error) {
+	if ai.useADAuthorizer {
+		glog.V(2).Infoln("No application insights key provided - using Azure GO SDK auth.")
+		return getMetricUsingADAuthorizer(ai, metricInfo)
+	}
+
+	glog.V(2).Infoln("Application insights key has been provided - using Application Insights REST API.")
+	return getMetricUsingAPIKey(ai, metricInfo)
+}
+
+func getMetricUsingADAuthorizer(ai AiAPIClient, metricInfo MetricRequest) (*insights.MetricsResult, error) {
+
+	authorizer, err := auth.NewAuthorizerFromEnvironmentWithResource(azureAdResource)
+	if err != nil {
+		glog.Errorf("unable to retrieve an authorizer from environment: %v", err)
+		return nil, err
+	}
+
+	metricsClient := insights.NewMetricsClient()
+	metricsClient.Authorizer = authorizer
```

Comment:
maybe `insights.NewMetricsClient(authorizer)`?
inject the dependency instead of setting it in the property.
it would be necessary if the metricsClient is an interface

Reply:
this is a limitation on the azure insights package, there is [no constructor for passing authorizer](https://github.com/Azure/azure-sdk-for-go/blob/6d20bdbae88c06c36d72eb512295417693bfdf4e/services/appinsights/v1/insights/metrics.go#L33). We can put this logic in the New() behind the AiApiClient when we move it to an interface.